From 2ff0c24a35f09e1d23e30d2802dd9826d4e28991 Mon Sep 17 00:00:00 2001
From: Christophe RICARD <christophe.ricard@gmail.com>
Date: Sat, 13 Feb 2016 16:15:24 +0100
Subject: [PATCH] tpm/st33zp24/spi: Remove nbr_dummy_bytes variable usage

References: bsc#1020645, fate#321435, fate#321507, fate#321600
Patch-mainline: v4.8-rc1
Git-commit: 9feaab5dd27c5caf13368055ced9d58a8302ff21

nbr_dummy_bytes variable could be easily replaced by phy->latency in
st33zp24_spi_send and st33zp24_spi_recv.

Signed-off-by: Christophe Ricard <christophe-h.ricard@st.com>
Reviewed-by: Jarkko Sakkinen <jarkko.sakkinen@linux.intel.com>
Signed-off-by: Jarkko Sakkinen <jarkko.sakkinen@linux.intel.com>
Acked-by: Michal Suchanek <msuchanek@suse.de>

There is some merge problem upstream. This commit is merged again as
604e57888499c149d0aca4b1ac39da07da040fcb partially reverting the following:

b83ada0caabd8d99329518b765f657ce1d0da24a tpm/st33zp24/spi: Use functions name with st33zp24_spi_
6e0219c751149c89f4d1f17dbd87086444933713 tpm/st33zp24/spi: Remove useless use of memcpy.
b637a366c2ca2327fbe042e8a214d5cc16272788 tpm/st33zp24/spi: Remove field spi_xfer from st33zp24_spi_phy
8d8d218e89e45717dfed6ac410e706f9c66d8255 tpm/st33zp24: Remove unneeded CONFIG_OF switches
10460bc1cfdea19dbab2289a63b4be0c874d29f2 tpm/st33zp24/spi: Improve st33zp24_spi_evaluate_latency
2c2b217a13ed11ff6d8f4583f9bee3a54e8f1034 tpm/st33zp24: Extend Copyright headers
fec60f2991c00504f6f862f0a6790da78c4ba904 tpm/st33zp24: Add support for acpi probing for i2c device.
160beb40d51b426c8557598f6f0f475dcd3e6d02 tpm: st33zp24: Add support for acpi probing for spi device.
6388b659533bf76ce62a3fb5f4f30a69bb65c098 tpm/st33zp24/i2c: Change xxx_request_resources header
add40d6df6babc43e587e9c8c6a2c1320bc78c98 tpm/st33zp24/spi: Change xxx_request_resources header

The whole series is merged again in

Git-commit: 604e57888499c149d0aca4b1ac39da07da040fcb tpm/st33zp24/spi: Remove nbr_dummy_bytes variable usage
Git-commit: 8a7450039a7b2a67f61f671a9aeb11b5176e2adc tpm/st33zp24/spi: Use functions name with st33zp24_spi_
Git-commit: d34306e28891257e3c4e8c172af19d70f84fa272 tpm/st33zp24/spi: Remove useless use of memcpy.
Git-commit: a5392e912031dff0b74a5d923c49c9839cea5bb3 tpm/st33zp24/spi: Remove field spi_xfer from st33zp24_spi_phy
Git-commit: 300796cdb5ba624031a5e1e5992e670eb457c231 tpm/st33zp24: Remove unneeded CONFIG_OF switches
Git-commit: 4ef2aa3c1bfe237cb349a52ed6f68e4cbfc5ccb9 tpm/st33zp24/spi: Improve st33zp24_spi_evaluate_latency
Git-commit: 8bb273f213a9ddd1c288baafcdfc02f02f0e471c tpm/st33zp24: Extend Copyright headers
Git-commit: 22eb90db936755e5d600a4561a0cc7a82c791eb9 tpm/st33zp24: Add support for acpi probing for i2c device.
Git-commit: 86ff205b8e7b5b85ef1eb5e13f92f37c40a7181a tpm: st33zp24: Add support for acpi probing for spi device.
Git-commit: 740ec346f32b3459ecf8ae27df76cfa60e94ead0 tpm/st33zp24/i2c: Change xxx_request_resources header
Git-commit: 4d8007ee2626066dbfd3917e813f7c2e47f96392 tpm/st33zp24/spi: Change xxx_request_resources header

These duplicate patches are not included but the IDs are noted here for
reference and fix lookup.
---
 drivers/char/tpm/st33zp24/spi.c | 20 +++++++++-----------
 1 file changed, 9 insertions(+), 11 deletions(-)

diff --git a/drivers/char/tpm/st33zp24/spi.c b/drivers/char/tpm/st33zp24/spi.c
index f974c94..74c5fcc 100644
--- a/drivers/char/tpm/st33zp24/spi.c
+++ b/drivers/char/tpm/st33zp24/spi.c
@@ -111,7 +111,7 @@ static int st33zp24_spi_send(void *phy_id, u8 tpm_register, u8 *tpm_data,
 			     int tpm_size)
 {
 	u8 data = 0;
-	int total_length = 0, nbr_dummy_bytes = 0, ret = 0;
+	int total_length = 0, ret = 0;
 	struct st33zp24_spi_phy *phy = phy_id;
 	struct spi_device *dev = phy->spi_device;
 	u8 *tx_buf = (u8 *)phy->spi_xfer.tx_buf;
@@ -133,14 +133,13 @@ static int st33zp24_spi_send(void *phy_id, u8 tpm_register, u8 *tpm_data,
 	memcpy(&tx_buf[total_length], tpm_data, tpm_size);
 	total_length += tpm_size;
 
-	nbr_dummy_bytes = phy->latency;
-	memset(&tx_buf[total_length], TPM_DUMMY_BYTE, nbr_dummy_bytes);
+	memset(&tx_buf[total_length], TPM_DUMMY_BYTE, phy->latency);
 
-	phy->spi_xfer.len = total_length + nbr_dummy_bytes;
+	phy->spi_xfer.len = total_length + phy->latency;
 
 	ret = spi_sync_transfer(dev, &phy->spi_xfer, 1);
 	if (ret == 0)
-		ret = rx_buf[total_length + nbr_dummy_bytes - 1];
+		ret = rx_buf[total_length + phy->latency - 1];
 
 	return st33zp24_status_to_errno(ret);
 } /* st33zp24_spi_send() */
@@ -157,7 +156,7 @@ static int st33zp24_spi_send(void *phy_id, u8 tpm_register, u8 *tpm_data,
 static int read8_reg(void *phy_id, u8 tpm_register, u8 *tpm_data, int tpm_size)
 {
 	u8 data = 0;
-	int total_length = 0, nbr_dummy_bytes, ret;
+	int total_length = 0, ret;
 	struct st33zp24_spi_phy *phy = phy_id;
 	struct spi_device *dev = phy->spi_device;
 	u8 *tx_buf = (u8 *)phy->spi_xfer.tx_buf;
@@ -171,18 +170,17 @@ static int read8_reg(void *phy_id, u8 tpm_register, u8 *tpm_data, int tpm_size)
 	memcpy(tx_buf + total_length, &data, sizeof(data));
 	total_length++;
 
-	nbr_dummy_bytes = phy->latency;
 	memset(&tx_buf[total_length], TPM_DUMMY_BYTE,
-	       nbr_dummy_bytes + tpm_size);
+	       phy->latency + tpm_size);
 
-	phy->spi_xfer.len = total_length + nbr_dummy_bytes + tpm_size;
+	phy->spi_xfer.len = total_length + phy->latency + tpm_size;
 
 	/* header + status byte + size of the data + status byte */
 	ret = spi_sync_transfer(dev, &phy->spi_xfer, 1);
 	if (tpm_size > 0 && ret == 0) {
-		ret = rx_buf[total_length + nbr_dummy_bytes - 1];
+		ret = rx_buf[total_length + phy->latency - 1];
 
-		memcpy(tpm_data, rx_buf + total_length + nbr_dummy_bytes,
+		memcpy(tpm_data, rx_buf + total_length + phy->latency,
 		       tpm_size);
 	}
 
-- 
2.10.2

